R/integrations.R

```diff
@@ -178,18 +214,37 @@ basicSeuratProc <- function(count.matrix, vars.to.regress=NULL, verbose=TRUE, do
 #'
 #' @export
 #'
-velocityInfoConos <- function(cms.list, con, n.odgenes = 2e3, verbose = TRUE, min.max.cluster.average.emat=0.2, min.max.cluster.average.nmat=0.05, min.max.cluster.average.smat=0.01) {
+velocityInfoConos <- function(cms.list, con, clustering=NULL, groups=NULL, n.odgenes=2e3, verbose=TRUE, min.max.cluster.average.emat=0.2, min.max.cluster.average.nmat=0.05, min.max.cluster.average.smat=0.01) {
   if (!requireNamespace("velocyto.R")) {
     stop("You need to install 'velocyto.R' package to be able to use this function")
   }
 
+  if(!is.null(clustering)) {
```

Comment:
I copied my implementation from the `findSubcommunities` function added recently by Rasmus, but I agree that what you are proposing here is much more robust.

Could the function be something like: 
```
setCellGroups <- function(con, clustering, groups) {
	if (is.null(groups)) {
  		if (is.null(clustering)) {
    		clustering <- names(clusters)[[1]]
  		}
  		if(is.null(clusters[[clustering]])) {
    		stop(paste("clustering",clustering,"doesn't exist, run findCommunity() first"))
  		}
  		cell.groups <- clusters[[clustering]]$groups
	} else if (!any(names(groups) %in% names(V(con$graph)))){
		stop("'groups' not defined for graph object.")
	} else {
		cell.groups <- groups
	}

	return(cell.groups)
```

Which we would then invoke by calling `groups <- setCellGroups(con, clustering, groups)`.
If it's okay I can add the function to the package.

Reply:
> I copied my implementation from the findSubcommunities function added recently by Rasmus, but I agree that what you are proposing here is much more robust.

Actually, for him I also made this comment, and now we have different implementation there. :)

Looks reasonable, but I'd change the style:

```r
parseCellGroups <- function(con, clustering, groups, parse.clusters=T) {
  if (!parse.clusters)
    return(groups)

  if (!is.null(groups)) {
    if (!any(names(groups) %in% names(V(con$graph))))
      stop("'groups' aren't defined for any of the cells.")

    return(groups)
  }

  if (is.null(clustering)) {
    if (length(con$clusters) > 0)
      return(con$clusters[[1]]$groups)

    stop("Either 'groups' must be provided or the conos object must have some clustering estimated")
  }
  if(is.null(clusters[[clustering]]))
    stop(paste("clustering",clustering,"doesn't exist, run findCommunity() first"))

  return(con$clusters[[clustering]]$groups)
}
```

Reply:
I added the function in fabac38. I implement it in my function and a couple more places, however, it would take significant more work to clean up the whole package. There are few functions (like `plotClusterBarplots`) that use their own checks with specific messages (and I don't want to mess with that). There is also the `getClusteringGroups`, which seems to be a similar function, I'm not sure if it makes sense to have them both.

Reply:
Thanks! I created an issue (https://github.com/hms-dbmi/conos/issues/48)